entsoe/entsoe.py

```diff
@@ -807,6 +861,54 @@ def query_imbalance_prices(self, country_code, start, end, psr_type=None):
         df = df.truncate(before=start, after=end)
         return df
 
+    @year_limited
+    @paginated
+    def query_contracted_reserve_prices(self, country_code, start, end,
+                                        type_marketagreement_type, psr_type = None):
+        """
+        Parameters
+        ----------
+        country_code : str
+        start : pd.Timestamp
+        end : pd.Timestamp
+        psr_type : str
+            filter query for a specific psr type
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        text = super(EntsoePandasClient, self).query_contracted_reserve_prices(
+            country_code = country_code, start = start, end = end,
+            type_marketagreement_type = type_marketagreement_type, psr_type = psr_type)
+        df = parse_contracted_reserve(text, TIMEZONE_MAPPINGS[country_code], "procurement_price.amount")
+        df = df.tz_convert(TIMEZONE_MAPPINGS[country_code])
+        return df
```

Comment:
This returns a Time Series, right? Best to add `df = df.truncate(before=start, after=end)`

Reply:
Done